oci/spec_opts_freebsd.go

```diff
@@ -0,0 +1,84 @@
+/*
+   Copyright The containerd Authors.
+
+   Licensed under the Apache License, Version 2.0 (the "License");
+   you may not use this file except in compliance with the License.
+   You may obtain a copy of the License at
+
+       http://www.apache.org/licenses/LICENSE-2.0
+
+   Unless required by applicable law or agreed to in writing, software
+   distributed under the License is distributed on an "AS IS" BASIS,
+   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+   See the License for the specific language governing permissions and
+   limitations under the License.
+*/
+
+package oci
+
+import (
+	"path/filepath"
+
+	specs "github.com/opencontainers/runtime-spec/specs-go"
+)
+
+// appendOSMounts modifies the mount spec to mount emulated Linux filesystems on FreeBSD,
+// as per: https://wiki.freebsd.org/LinuxJails
+func appendOSMounts(s *Spec, os string) error {
+	// No-op for FreeBSD containers
+	if os != "linux" {
+		return nil
+	}
+	var mounts []specs.Mount
+	var (
+		haveProc  = false
+		haveDevFd = false
+	)
+	/* If the mounts exist, we modify them in place, to keep the current order
+	   otherwise we add them below the for-loop. The nosuid noexec options are
+	   for consistency with Linux mounts: on FreeBSD it is by default impossible
+	   to execute anything from these filesystems.
+	*/
+	for _, m := range s.Mounts {
+		path := filepath.Clean(m.Destination)
+		if path == "/proc" {
+			mounts = append(mounts, specs.Mount{
+				Destination: "/proc",
+				Type:        "linprocfs",
+				Source:      "linprocfs",
+				Options:     []string{"nosuid", "noexec"},
+			})
+			haveProc = true
+			continue
+		}
+		if path == "/dev/fd" {
+			m.Options = append(m.Options, "linrdlink")
+			haveDevFd = true
+		}
+		mounts = append(mounts, m)
+	}
+	if !haveProc {
+		mounts = append(mounts, specs.Mount{
+			Destination: "/proc",
+			Type:        "linprocfs",
+			Source:      "linprocfs",
+			Options:     []string{},
```

Comment:
In the in-place case above, we're adding "nosuid" and "noexec" as options, but in this case we're not.  Is there any particular reason for the difference?

Reply:
No reason, we should have nosuid & noexec in both ases.